test/new-e2e/tests/windows/common/version.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package common
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+)
+
+// GetFileVersion gets the file version information from the versioninfo resource
+// of the specified file.  It returns as a slice of strings, each element being part
+// of the version (major, minor, patch, build)
+//
+// // for the remote file, it uses the powershell version information command
+func GetFileVersion(host *components.RemoteHost, remoteFileName string) ([]string, error) {
+	pscommand := fmt.Sprintf(`(Get-Item "%s").VersionInfo.FileVersion`, remoteFileName)
+	remoteversion, err := host.Execute(pscommand)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get version: %w", err)
+	}
+	return strings.Split(strings.TrimSpace(remoteversion), "."), nil
+}
+
+// GetExpectedVersion gets the expected version from the most recent tag
+// returns in the form of slice of strings, each element being a part of the version
+// (major, minor, patch)
+//
+// expectedVersion can be empty; if so, then it uses `git` to determine the
+// expected version from the most recent tag.  It also then assumes that
+// git is in the path
+func GetExpectedVersion(expectedVersion string) ([]string, error) {
+	if expectedVersion == "" {
+		// this needs to be executed locally
+		output, err := exec.Command("git", "describe", "--tags", "--candidates=50", "--match", "[0-9]*", "--abbrev=7").Output()
```

Comment:
yes.  because the initial consumer is not the agent; so there's no agent command.  The initial consumer of this is the RUM IIS injector.  But, I added it here because it seemed it could be generally useful.  

This is also why you can pass the `expectedVersion` as an argument
1) if you know the version (specifically, the upgrade case where you're first installing an old version that's not tied to the git branch)
2) if you want to get the from version from some external source that does not require shelling out.